mlflow/models/utils.py

```diff
@@ -950,12 +987,66 @@ def _is_scalar(x):
             )
     if input_schema.is_tensor_spec():
         return _enforce_tensor_schema(pf_input, input_schema)
+    elif HAS_PYSPARK and isinstance(original_pf_input, SparkDataFrame):
+        return _enforce_pyspark_dataframe_schema(
+            original_pf_input, pf_input, input_schema, flavor=flavor
+        )
     elif input_schema.has_input_names():
         return _enforce_named_col_schema(pf_input, input_schema)
     else:
         return _enforce_unnamed_col_schema(pf_input, input_schema)
 
 
+def _enforce_pyspark_dataframe_schema(
+    original_pf_input: SparkDataFrame,
+    pf_input_as_pandas,
+    input_schema: Schema,
+    flavor: Optional[str] = None,
+):
+    """
+    Enforce that the input PySpark DataFrame conforms to the model's input schema.
+
+    This function creates a new DataFrame that only includes the columns from the original
+    DataFrame that are declared in the model's input schema. Any extra columns in the original
+    DataFrame are dropped.Note that this function does not modify the original DataFrame.
+
+    :param original_pf_input: Original input PySpark DataFrame.
+    :param pf_input_as_pandas: Input DataFrame converted to pandas.
+    :param input_schema: Expected schema of the input DataFrame.
+    :param flavor: Optional model flavor. If specified, it is used to handle specific behaviors
+                   for different model flavors. Currently, only the '_FEATURE_STORE_FLAVOR' is
+                   handled specially.
+    :return: New PySpark DataFrame that conforms to the model's input schema.
+    """
+    if SparkDataFrame is None:
+        raise ValueError("PySpark is not installed. Cannot handle a PySpark DataFrame.")
```

Comment:
Please change this to raise an MlflowException instead of a ValueError. We're specifically generating this exception and we can declare specificity that this is effectively an environment configuration issue (trying to use a SparkDF input type when Spark isn't installed (which shouldn't even be possible, but it's good to have a consistent error handling here just in case)